scm/scm_manager.go

```diff
@@ -0,0 +1,115 @@
+package scm
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+
+	"github.com/beatlabs/github-auth/app"
+	"github.com/beatlabs/github-auth/key"
+	"github.com/google/go-github/v45/github"
+	"go.uber.org/zap"
+)
+
+const installationAPI = "https://api.github.com/app/installations"
+
+// GithubConfig keeps parameters of the GitHub app.
+type GithubConfig struct {
+	appID     string
+	clientID  string
+	secret    string
+	keyPath   string
+	appConfig *app.Config
+}
+
+// SCMManager keeps provider-specific configs (currently only for GitHub)
+// and a map of scm clients for each course.
+type SCMManager struct {
+	Scms         *Scms
+	githubConfig *GithubConfig
+}
+
+// newGitHubConfig creates a new configuration for GitHub app
+func newGitHubConfig(appID, appKey, appSecret, appKeyFile string) *GithubConfig {
+	return &GithubConfig{
+		appID:    appID,
+		clientID: appKey,
+		secret:   appSecret,
+		keyPath:  appKeyFile,
+	}
+}
+
+// valid ensures that all configuration fields are not empty
+func (conf *GithubConfig) valid() bool {
+	return conf.appID != "" && conf.keyPath != "" &&
+		conf.clientID != "" && conf.secret != ""
+}
+
+// NewSCMManager creates base client for the QuickFeed GitHub Application. This client
+// cannto access organization-related GitHub API and is used to derive installation
```

Comment:
Typo: cannot. But suggest to rephrase:
`This client can be used to install API clients for each course organization.`